File: javaCourse/Lab-3/CosmeticDepartment.java
package com.scrubele;

import java.util.Comparator;
import java.util.LinkedList;
import java.util.List;

public class CosmeticDepartment {
    private String name;
    private String addressOfShop;
    private int floor;
    private List<BeatyProduct> productList;

    public CosmeticDepartment(String name, String addressOfShop, int floor) {
        this.name = name;
        this.addressOfShop = addressOfShop;
        this.floor = floor;
        this.productList = new LinkedList<>();
    }

    public void addProduct(BeatyProduct beatyProduct) {
        productList.add(beatyProduct);
    }

    public void displayProducts(List<BeatyProduct> productList) {
        productList.sort(Comparator.comparing(BeatyProduct::getPrice).reversed());
        System.out.println("We can offer you:");
        for (BeatyProduct product : productList) {
            System.out.println(product.toString());
        }
        System.out.println();
    }


    public List<BeatyProduct> findAvailableProducts(boolean choice, double money) {
        List<BeatyProduct> avialableForCustomerProducts = new LinkedList<>();
        int countOfProducts = 0;
        double sum = 0;
        if (choice == true) {
            productList.sort(Comparator.comparing(BeatyProduct::getPrice).reversed());

            for (BeatyProduct product : productList) {
                if ((product.getQuantity() > 0) && (product.getPrice() <= money)) {
                    avialableForCustomerProducts.add(product);
                    countOfProducts++;
                }
            }

        } else {

            productList.sort(Comparator.comparing(BeatyProduct::getPrice));
            while (sum < money) {
                for (BeatyProduct product : productList) {
                    if (product.getQuantity() > 0) sum += product.getPrice();
                    if (sum < money) {
                        avialableForCustomerProducts.add(product);
                        countOfProducts++;
                    }
                }
            }
        }

        if (countOfProducts > 0) {
            displayProducts(avialableForCustomerProducts);
        } else {
            System.out.println("Ohh, no... You can not buy any product. Sorry :(");
        }
        return avialableForCustomerProducts;
    }

    @Override
    public String toString() {
        return "CosmeticDepartment{" +
                "name='" + name + '\'' +
                ", addressOfShop='" + addressOfShop + '\'' +
                ", floor=" + floor +
                ", productList=" + productList;
    }

    public List<BeatyProduct> getProductList() {
        return productList;
    }
}

File: pythonCourse/Lab2/CosmeticDepartment.py

class CosmeticDepartment(object):
    __name, __address_of_shop, __floor, __product_list = "", "", 0, []

    def __init__(self, name, address_of_shop, floor):
        """Return a Customer object with 4 attributes: *name*, *price*,
        *type*,*quantity*"""
        self.__name = name
        self.__address_of_shop = address_of_shop
        self.__floor = floor
        self.__product_list = []

    def add_product(self, beauty_product):
        self.__product_list.append(beauty_product)

    def to_string(self):
        """Print object."""
        print("Name: " + self.__name + ", address: " + self.__address_of_shop +
              ", type: " + str(self.__floor))

    def find_available_products(self, choice, money):
        available_for_customer_products = []
        count_of_products = 0
        sum = 0
        if choice == 1:
            self.__product_list.sort()
            for beauty_product in self.__product_list:
                if beauty_product.quantity > 0 and beauty_product.price < money:
                    available_for_customer_products.append(beauty_product)
                    count_of_products += 1

        else:
            self.__product_list.sort(reverse=True)
            for beauty_product in self.__product_list:
                if beauty_product.quanity > 0:
                    sum += beauty_product.price
                if sum < money:
                    available_for_customer_products.append(beauty_product)
                    count_of_products += 1
        if count_of_products > 0:
            self.display_products(available_for_customer_products)
        else:
            print("Ohh, no... You can not buy any product. Sorry :(")

        return available_for_customer_products

    def get_list(self):
        return self.__product_list

    @staticmethod
    def display_products(p_list):
        p_list.sort(reverse=True)
        for beauty_product in p_list:
            print(beauty_product)

